framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -993,6 +993,94 @@ void OpenXrReplayConsumerBase::Process_xrEnumerateEnvironmentBlendModes(
     }
 }
 
+void OpenXrReplayConsumerBase::Process_xrEnumerateDisplayRefreshRatesFB(
+    const ApiCallInfo&        call_info,
+    XrResult                  returnValue,
+    format::HandleId          session,
+    uint32_t                  displayRefreshRateCapacityInput,
+    PointerDecoder<uint32_t>* displayRefreshRateCountOutput,
+    PointerDecoder<float>*    displayRefreshRates)
+{
+    XrResult  replay_result = XR_SUCCESS;
+    XrSession in_session    = MapHandle<OpenXrSessionInfo>(session, &CommonObjectInfoTable::GetXrSessionInfo);
+    uint32_t* out_displayRefreshRateCountOutput =
+        displayRefreshRateCountOutput->IsNull()
+            ? nullptr
+            : displayRefreshRateCountOutput->AllocateOutputData(1, static_cast<uint32_t>(0));
+    float* out_displayRefreshRates = nullptr;
+    if (!displayRefreshRates->IsNull() && (returnValue == XR_SUCCESS || returnValue == XR_ERROR_SIZE_INSUFFICIENT))
+    {
+        auto pfn_enum_dis_refresh_rates_fb = GetInstanceTable(in_session)->EnumerateDisplayRefreshRatesFB;
+
+        // First get the actual count for the replay
+        uint32_t replay_count = 0;
+        replay_result         = pfn_enum_dis_refresh_rates_fb(in_session, 0, &replay_count, nullptr);
+
+        if (replay_result == XR_SUCCESS || replay_result == XR_ERROR_SIZE_INSUFFICIENT)
+        {
+            if (replay_count < displayRefreshRateCapacityInput)
+            {
+                GFXRECON_LOG_FATAL(
+                    "xrEnumerateDisplayRefreshRatesFB failed to find as many items during replay as during capture");
+                return;
+            }
+
+            // Allocate the blend mode array and get all the values
+            std::vector<float> display_refresh_rates;
+            display_refresh_rates.resize(replay_count);
```

Comment:
You could do this in one step with `std::vector<float> display_refresh_rates(replay_count);`

Reply:
And use `display_refresh_rates.size()` in the call to `pfn_enum_dis_refresh_rates_fb` instead of `replay_count`